### PR TITLE
Add support for providing optional logits processors to HuggingFace text-only LLM and MLLM models for output control 

### PR DESCRIPTION
**Pull request by HJJ256**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## New Package?

Did I fill in the `tool.llamahub` section in the `pyproject.toml` and provide a detailed README.md for my new integration or package?

- [ ] Yes
- [x] No

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## How Has This Been Tested?
Tested locally on my server. I was able to add a [SynCode](https://github.com/uiuc-focal-lab/syncode) JSON logits processor to the updated models and make them generate a strict JSON output. Previously all models (I tested with the small models) were only generating markdowns for me. Even json was generated as a markdown like this

\`\`\`json
{
    "key": "value"
} 
\`\`\`

Especially in case of small models: The models misbehave a lot and try to output extra or differently formatted information. Enforcing a grammar using logits processors helps in controlling the output of the model in a more deterministic way.

Your pull-request will likely not be merged unless it is covered by some form of impactful unit testing.

- [ ] I added new unit tests to cover this change
- [ ] I believe this change is already covered by existing unit tests

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Review by logan-markewich — APPROVED** [review]

**Comment by HJJ256** [comment]
Hi @logan-markewich 
Are there any updates on the release for this feature? Is there anything required from my end?

Thanks

**Comment by HJJ256** [comment]
Please review this PR again, have added the same logic for text-only LLMs as well
